packages/core/lib/testing/testrunner.js

```diff
@@ -49,231 +50,164 @@ function TestRunner(options = {}) {
-      }
-
-      self.logger.log("\n    Events emitted during test:");
-      self.logger.log("    ---------------------------");
-      self.logger.log("");
-
-      logs.forEach(function(log) {
-        var event = self.known_events[log.topics[0]];
+  function printEvent(decoding) {
+    const anonymousPrefix = decoding.kind === "anonymous" ? "<anonymous> " : "";
+    const className = decoding.class.typeName;
+    const eventName = decoding.abi.name;
+    const fullEventName = anonymousPrefix + `${className}.${eventName}`;
+    const eventArgs = decoding.arguments
+      .map(({ name, indexed, value }) => {
+        let namePrefix = name ? `${name}: ` : "";
+        let indexedPrefix = indexed ? "<indexed> " : "";
+        let displayValue = util.inspect(
+          new Codec.Format.Utils.Inspect.ResultInspector(value),
+          {
+            depth: null,
+            colors: true,
+            maxArrayLength: null,
+            breakLength: Infinity,
+            compact: true
+          }
+        );
+        return namePrefix + indexedPrefix + displayValue;
+      })
+      .join(", ");
+    return `${fullEventName}(${eventArgs})`;
+  }
 
-        if (event == null) {
-          return; // do not log anonymous events
-        }
+  const logs = await this.decoder.events({
+    //NOTE: block numbers shouldn't be over 2^53 so this
+    //should be fine, but should change this once decoder
+    //accepts more general types for blocks
+    fromBlock: this.currentTestStartBlock.toNumber()
+  });
 
-        var types = event.abi_entry.inputs.map(function(input) {
-          return input.type;
-        });
+  if (logs.length === 0) {
+    this.logger.log("    > No events were emitted");
+    return;
+  }
 
-        var values;
-        try {
-          values = abi.decodeLog(
-            event.abi_entry.inputs,
-            log.data,
-            log.topics.slice(1) // skip topic[0] for non-anonymous event
-          );
-        } catch (_) {
-          //temporary HACK until we're using the new decoder
-          self.logger.log(`    Warning: event decoding failed`);
-          self.logger.log(
-            `    (This may be due to multiple events with same signature`
-          );
-          self.logger.log(`    or due to unsupported data types)`);
-          return;
+  this.logger.log("\n    Events emitted during test:");
+  this.logger.log("    ---------------------------");
+  this.logger.log("");
+
+  for (const log of logs) {
+    switch (log.decodings.length) {
+      case 0:
+        this.logger.log(`    Warning: Could not decode event!`);
+        break;
+      case 1:
+        this.logger.log("    " + printEvent(log.decodings[0]));
+        break;
+      default:
+        this.logger.log(`    Ambiguous event, possible interpretations:`);
+        for (const decoding of log.decodings) {
+          this.logger.log(`      Could be:`); //6 spaces to indent a little
+          this.logger.log("      " + printEvent(decoding));
```

Comment:
And just make this a bulleted list:
```suggestion
          this.logger.log("    - " + printEvent(decoding));
```

Reply:
Before:
<img width="855" alt="Screen Shot 2019-11-27 at 1 24 14 PM" src="https://user-images.githubusercontent.com/151065/69749884-6f74c680-1119-11ea-899a-bdb9a43651f0.png">

After:
<img width="873" alt="Screen Shot 2019-11-27 at 1 25 24 PM" src="https://user-images.githubusercontent.com/151065/69749889-71d72080-1119-11ea-9b9b-94a041cd5170.png">

Looks better, I think

Reply:
Hm, might want to clean this up so multiple events are easier to read (maybe a blank line in between would suffice?):
<img width="909" alt="Screen Shot 2019-11-27 at 2 25 59 PM" src="https://user-images.githubusercontent.com/151065/69753562-e615c200-1121-11ea-83fd-a5a002551617.png">

Reply:
I ended up adding the blank line, and also used `*` instead of `-` for clarity.